pkg/middleware/validation.go

```diff
@@ -45,31 +48,50 @@ func NewValidationMiddleware(validatorHook string) mux.MiddlewareFunc {
 	}
 }
 
+func maybeLogValidationReason(valid ValidationResponse, r *http.Request, mod string, version string) {
```

Comment:
this signature can be slightly improved, since you don't need to use any underlying variables from an http.Response or a ValidationResponse: 

`func maybeLogValidationReason(ctx context.Context, msg, mod, version string)  {...}` -- or maybe even take a `log.Entry` instead of context, but happy either way.

Reply:
https://github.com/gomods/athens/pull/1517/commits/12dba10897b22c735db29b02d50fd64a85bba2ea